src/graphics/programlib/programlib_phong.js

```diff
@@ -192,7 +229,18 @@ pc.programlib.phong = {
         //////////////////////////////
         // GENERATE FRAGMENT SHADER //
         //////////////////////////////
+        var fshader;
         code = getSnippet(device, 'fs_precision');
+
+        if (options.customPS) {

```

Comment:
This one is internal again. Don't look at that file :D Public API is in scene_phongmaterial